File: 2502-sort-the-people/sort-the-people.py
class Solution:
    def sortPeople(self, names: List[str], heights: List[int]) -> List[str]:
        n = len(names)
        helper = list(range(n))
        for i in range(n):
            helper[i] = (heights[i], names[i])
        helper = sorted(helper, reverse=True)
        answer = list()
        for height, name in helper:
            answer.append(name)
        return answer
        

File: 2502-sort-the-people/sort-the-people.cpp
class Solution {
public:
    vector<string> sortPeople(vector<string>& names, vector<int>& heights) {
        int n = names.size();
        vector<pair<string, int>> helper(n);
        for (int i = 0; i < n; i++){
            helper[i] = {names[i], heights[i]};
        }
        sort(helper.begin(), helper.end(), [](pair<string, int> &l, pair<string, int> &r){
            return l.second > r.second;
        });
        vector<string> answer(n);
        for (int i = 0; i < n; i++){
            answer[i] = helper[i].first;
        }
        return answer;
    }
};